build.py

```diff
@@ -165,10 +168,18 @@ def check_mlx():
 
 def generate_extra_mk():
```

Comment:
I am a bit concerned that `generate_extra_mk()` is called only during configure_dpdk(). If DPDK is already built (e.g., the docker image for `container_build.py` comes with pre-built DPDK) the extra.mk file will never be updated.

One potential solution to address this issue is that we separate `core/extra_dpdk.mk` from `core/extra.mk`, both of which are included by `core/Makefile` (.gitignore should include the both, of course), so that `core/extra.mk` can be updated by `build_bess()`.